RxSwift/Observables/Sample.swift

```diff
@@ -119,10 +123,12 @@ final private class SampleSequenceSink<Observer: ObserverType, SampleType>
 final private class Sample<Element, SampleType>: Producer<Element> {
     fileprivate let source: Observable<Element>
     fileprivate let sampler: Observable<SampleType>
-
-    init(source: Observable<Element>, sampler: Observable<SampleType>) {
+    fileprivate let defaultValue: Element?
```

Comment:
add a `defaultValue` argument to `SampleSequenceSink` and pass it inside. Doing `parent.parent` doesn't make much sense in this case.